### DISCUSSION TITLE
LUKS with systemd-cryptenroll for root fs and additional volumes?

### DISCUSSION THREAD

**Discussion by jmpolom** [rancher/elemental#1428] [General]
Is it possible to use LUKS encrypted volumes with elemental and have them auto unlock at boot with systemd-cryptenroll? Has anyone tried this successfully?

**Comment by kkaempf** [comment]
Encrypted volumes (esp. encrypted rootfs) is on our backlog. No further investigations have been done yet.

**Comment by kkaempf** [comment]
I'd be interested in how you envision providing the decryption passphrase to a machine at boot 🤔

**Reply by jmpolom** [reply]
Password unlock would clearly only be for hands on rescue/recovery operations. It obviously is not suitable for headless operations which is why I mentioned systemd-cryptenroll. Do you know what systemd-cryptenroll is?
